### PR TITLE
Switch Inotify to irmin-watcher

### PR DESCRIPTION
**Pull request by patricoferris**
Thanks for _slips_ @panglesd ! 

Irmin-watcher is a portable, filesystem notification library. On linux it uses inotify and on macOS it uses fsevents. It has a portable, if slow, polling mode too.

This makes slipshow installable on macOS \o/

(( Sorry about all of the formatting that's been sucked into this PR :S -- feel free to cherry pick just the meaninful changes ))

### CODE REVIEW

**Comment by panglesd** [comment]
Thanks a lot, this is great!

I had the plan to create a multiplatform filesystem watcher but independent of irmin (`irmin-watcher` might not have `irmin` as a dependency, but if I understand correctly `irmin` influences `irmin-watcher`'s API).
I would be based on `inotify`, `fsevents` but also `kqueue` (for Mac but also for BSDs). I also planned to add some higher level API and some docs: the topic of file watching is more difficult than I though, with multiplatform, atomic writes, emacs, vim and vscode all saving in different ways...
I definitely made progress in this direction, but it'll still require some time.

Having your PR ready _now_ is really nice for all Mac users! Thanks!

File: compiler/src/bin/native/serve.ml [hunk 1]
```diff
@@ -51,7 +54,7 @@ let html_source =
 </body>
 </html>
   |html}
-    [%blob "compiler/src/bin/native/client/client.bc.js"]
+    [%blob "client/client.bc.js"]
```
Comment (panglesd) [inline compiler/src/bin/native/serve.ml:57]: Hmm it seems to work in both situation, is it some `blob` ppx magic? Anyway, the short form is better.

File: compiler/data/data_contents.ml
Comment (panglesd) [inline compiler/data/data_contents.ml:1]: I think you have ocamlformatted this file, but there is a `.ocamlformat` in the directory which should have disabled that...
Can you revert the change?

File: compiler/src/bin/native/dune [hunk 2]
```diff
@@ -4,12 +4,6 @@
...
- (libraries
-  slipshow
-  fpath
-  lwt
-  inotify.lwt
-  dream
-  ; bos
-  ))
+ (preprocessor_deps
+  (file client/client.bc.js))
```
Comment (panglesd) [inline compiler/src/bin/native/dune:8]: From the dune manual, `(file <filename>)` and `<filename>` is the same. What is the reason to prefer on to the other?

**Review by panglesd — APPROVED** [review]
I'm excited to cut a release with your changes! This will fix #56 

I have some questions, mostly on changes unrelated to the core of this PR.

Thanks again!

**Comment by panglesd** [comment]
(The CI issue seems unrelated to this. I'll need to fix to do a release...)

File: compiler/src/bin/native/serve.ml (hunk 1 above)
Comment (patricoferris) [reply compiler/src/bin/native/serve.ml:57]: The long form _doesn't_ work when you vendor (in the dune sense) slipshow ^^"

File: compiler/src/bin/native/dune (hunk 2 above)
Comment (patricoferris) [reply compiler/src/bin/native/dune:8]: Oh that was from testing the ppx_blob issue I mentioned above

File: compiler/src/bin/native/serve.ml (hunk 1 above)
Comment (panglesd) [reply compiler/src/bin/native/serve.ml:57]: Ah yes! I understand. Thanks for the explanation :)
